src/Package/Impl/ProjectSystem/Commands/SourceFilesCommand.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.ProjectSystem.Designers;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.R.Package.Commands;
+using Microsoft.VisualStudio.R.Package.Repl.Commands;
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem.Commands {
+    [ExportCommandGroup("AD87578C-B324-44DC-A12A-B01A6ED5C6E3")]
+    [AppliesTo("RTools")]
+    [OrderPrecedence(200)]
+    internal sealed class SourceFilesCommand : ICommandGroupHandler {
+        public CommandStatusResult GetCommandStatus(IImmutableSet<IProjectTree> nodes, long commandId, bool focused, string commandText, CommandStatus progressiveStatus) {
+            if (commandId == RPackageCommandId.icmdSourceSelectedFiles && !nodes.IsFolder() && nodes.GetSelectedNodesPaths().Count() > 0) {
```

Comment:
I guess we can filter this out in the handler. Probably not worth enumerating and checking if user actually selected 3 folders. Most common case is when they click on a folder node and that's where we want to hide the command